test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTests.cs

```diff
@@ -0,0 +1,344 @@
+        <packageSource key=""contoso"">
+            <namespace id=""moreStuff"" />
+        </packageSource>
+        <packageSource key=""privateRepository"">
+            <namespace id=""private*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var configuration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(settings);
+            Assert.True(configuration.IsNamespacesEnabled);
+            configuration.Namespaces.Should().HaveCount(3);
+
+            var packageSourcesMatchFull1 = configuration.GetConfiguredPackageSources("stuff");
+
+            Assert.Equal(1, packageSourcesMatchFull1.Count);
+            Assert.Equal("nuget.org", packageSourcesMatchFull1.First());
+
+            var packageSourcesMatchPartial1 = configuration.GetConfiguredPackageSources("stu");
+            Assert.Null(packageSourcesMatchPartial1);
+
+            var packageSourcesMatchFull2 = configuration.GetConfiguredPackageSources("moreStuff");
+            Assert.Equal(1, packageSourcesMatchFull2.Count);
+            Assert.Equal("contoso", packageSourcesMatchFull2.First());
+
+            var packageSourcesMatchPartial2 = configuration.GetConfiguredPackageSources("PrivateTest");
+            Assert.Equal(1, packageSourcesMatchPartial2.Count);
+            Assert.Equal("privaterepository", packageSourcesMatchPartial2.First());
+
+            var packageSourcesNoMatch = configuration.GetConfiguredPackageSources("random");
+            Assert.Null(packageSourcesNoMatch);
+        }
+
+        [Fact]
+        public void SearchTree_WithMultipleSourcesMultiparts()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageNamespaces>
+        <packageSource key=""PublicRepository""> 
+            <namespace id=""Contoso.Public.*"" />
+            <namespace id=""Contoso.Opensource.*"" />
+        </packageSource>
+        <packageSource key=""PrivateRepository"">
+            <namespace id=""Contoso.Opensource"" />
+        </packageSource>
+        <packageSource key=""SharedRepository"">
+            <namespace id=""Contoso.MVC*"" />
+        </packageSource>
+        <packageSource key=""MetaRepository"">
+            <namespace id=""meta.cache*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var configuration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(settings);
+            Assert.True(configuration.IsNamespacesEnabled);
```

Comment:
nit: if you want to use fluent, you can write `configuration.IsNamespacesEnabled.Should().BeTrue()`

Reply:
Sure, but since I already started with `Assert.xxx` pattern converting them could take some time. I'll try to use fluent assertion more next time.

`restoreResult.Success.Should().BeTrue(because: restoreResult.AllOutput);`

Reply:
Yep, it's just a nit, I don't care too much :)